### PR TITLE
Split CSS vars for padding values

### PR DESCRIPTION
**Pull request by mdo**
Make these few components consistent with where we're heading with other components. Had to add some new Sass variables to handle the dropdown-header element, but not a huge deal. This ensures we can drop the combined variable in v6 when we're ready.

### CODE REVIEW

File: scss/_dropdown.scss [hunk 1]
```diff
@@ -37,16 +38,18 @@
...
   --#{$variable-prefix}dropdown-link-disabled-color: #{$dropdown-link-disabled-color};
-  --#{$variable-prefix}dropdown-item-padding: #{$dropdown-item-padding-y $dropdown-item-padding-x};
+  --#{$variable-prefix}dropdown-item-padding-x: #{$dropdown-item-padding-x};
+  --#{$variable-prefix}dropdown-item-padding-y: #{$dropdown-item-padding-y};
   --#{$variable-prefix}dropdown-header-color: #{$dropdown-header-color};
-  --#{$variable-prefix}dropdown-header-padding: #{$dropdown-header-padding};
-  // scss-docs-end dropdown-css-vars
+  --#{$variable-prefix}dropdown-header-padding-x: #{$dropdown-header-padding-x};
+  --#{$variable-prefix}dropdown-header-padding-y: #{$dropdown-header-padding-y};
+  // scss-docs-end dropdown-css-vary
```
Comment (julien-deramond) [inline scss/_dropdown.scss:46]:
```suggestion
  // scss-docs-end dropdown-css-vars
```

File: scss/_variables.scss [hunk 2]
```diff
@@ -1150,7 +1150,9 @@ $dropdown-item-padding-y:           $spacer * .25 !default;
 $dropdown-item-padding-x:           $spacer !default;
 
 $dropdown-header-color:             $gray-600 !default;
-$dropdown-header-padding:           $dropdown-padding-y $dropdown-item-padding-x !default;
+$dropdown-header-padding-x:         $dropdown-item-padding-x !default;
+$dropdown-header-padding-y:         $dropdown-padding-y !default;
+$dropdown-header-padding:           $dropdown-header-padding-y $dropdown-header-padding-x !default;
```
Comment (julien-deramond) [inline scss/_variables.scss:1155]: This var becomes unused so can be dropped.
```suggestion
```

**Review by julien-deramond — COMMENTED** [review]
It clearly brings more consistency. I appreciate this change :)

File: scss/_variables.scss (hunk 2 above)
Comment (ffoodd) [reply scss/_variables.scss:1155]: I think we should deprecate it but we can't drop it IMHO, could cause compilation issue for folks using it in their custom build.

Comment (julien-deramond) [reply scss/_variables.scss:1155]: Yes my bad. You're totally right.

**Review by ffoodd — APPROVED** [review]
Looks good! I think we can deprecate `$dropdown-header-padding` indeed, as pointed out by @julien-deramond 👌

**Pull request merged**